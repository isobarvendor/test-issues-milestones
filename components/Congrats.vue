<template>
  <client-only>
    <div class="congrats" v-if="redeemPrize">

        <h3> Congratulations! </h3>
            <div class="input-container">
                <img :src="redeemPrize.imgUrl" />
            </div>
            <div class="desc">
                <p>Reward has been successfully redeemed<br/>and added to your wallet!</p>
            </div>
            <div class="button-container">
                <a  class="button" href="#" onclick="window.history.back()">back to reward list</a>
            </div>
    </div>
</client-only>
</template>

<script>
export default {
  name: "Congrats",
  components: {
  },
  data() {
    return {
    };
  },
  props: {
    data: null,
  },
  computed:{
     redeemPrize(){
      return this.$store.state.redeemPrize ? this.$store.state.redeemPrize.redeemedPrize : null;
    }
  },
  methods:{
  },
  beforeMount() {},
  mounted(){
    if(!this.$store.state.redeemPrize){
      location.href="/";
    }
  },
  beforeDestroy(){
    //this.$store.commit('SET_REDEEM_PRIZE',null);
  }
};
</script>

<style lang="scss">
.congrats{
    display: flex;
    align-items: center;
    flex-direction: column;
    .theme--light.v-application{
        background: none !important;
    }
    .v-application--wrap{
        min-height:auto !important;
        display: flex;
        flex-direction: row;
    }
    .v-application .primary--text{
        caret-color: #ffffff !important;
    }
    .theme--light.v-input{
        width: 25px;
        margin-right: 15px;
        &:last-child{
            margin-right: 0px;
        }

    }

    .theme--light.v-input input{
        color: white;
        text-align: center;
    }
    .theme--light.v-text-field > .v-input__control > .v-input__slot:before{
        border-color: #585858;
    }
    .mdi:before, .mdi-set{
        color: white;
    }
    .input-container{
        margin-top: 30px;
        max-width: 250px;
        justify-content: center;
        align-items: center;
        display:flex;
        .value{
            font-size: 50px;
            span{
                font-size: 13px;
                display: block;
            }
        }
        img{
          width: 100%;
        }
    }
    .desc{
        p{
            margin-top: 30px;
            text-align: center;
            &.code{
                color: #585858;
                font-size: 14px;
            }
        }

    }
    @media only screen and (max-width: 1199px) {

    }
}
</style>
