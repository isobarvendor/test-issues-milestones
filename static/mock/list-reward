[ {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   }
]
