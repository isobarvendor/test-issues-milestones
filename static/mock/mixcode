{
                      "programId": 453269,
                      "code": "3WTD90GE5",
                      "lot": {
                        "lotId": 4455843436,
                        "lotName": "Coke 500ml PET",
                        "activateDate": 1611208800000,
                        "inactivateDate": 1627603200000,
                        "active": true,
                        "expired": false,
                        "bevProdPkg": {
                          "trademarkCd": "05",
                          "trademarkName": "Coca-Cola",
                          "brandCd": "001",
                          "brandName": "Coca-Cola",
                          "bevProdCd": "0001",
                          "bevProdName": "Coca-Cola",
                          "caffeinated": true,
                          "carbonated": true,
                          "calorieCategory": {
                            "code": 1,
                            "description": "Regular"
                          }
                        },
                        "releaseId": 494494,
                        "releaseName": "Coke PET",
                        "organizationId": 157168,
                        "organizationName": "Harish",
                        "realtimeCode": false,
                        "testRelease": false,
                        "pointValue": 0,
                        "bevProdOptions": []
                      },
                      "redeemed": false,
                      "actualCode": "3WTD90GE5"
                    }
