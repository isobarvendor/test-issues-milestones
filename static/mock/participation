{"id":16,"questionId":1,"answerId":3,"participationId":17}
