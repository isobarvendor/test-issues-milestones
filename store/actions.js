
import GeneralAPI from '../api/general';

import {LOGIN, SIGNUP, GET_ACCOUNT} from './action_types';

export default {
  [LOGIN]: ({ commit, state, getters }, data) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.login(data)
        .then(response => {
          commit('SET_TOKEN', response.data.token);
          sessionStorage.token=response.data.token;
          return resolve(messages);
        })
        .catch(error => {

          return reject(error);
        });
      }
    });
  },
  [GET_ACCOUNT]: ({ commit, state, getters }, token) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.getAccount(token)
        .then(response => {

          commit('SET_LOGIN_ACCOUNT', response.data);
          return resolve(messages);
        })
        .catch(error => {

          return reject(error);
        });
      }
    });
  },
  [SIGNUP]: ({ commit, state, getters }, data) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.signup(data)
        .then(messages => {
          //commit('SET_LOGIN_ACCOUNT', messages);
          return resolve(messages);
        })
        .catch(error => {
          console.error(error);
          return reject(error);
        });
      }
    });
  },


  /*[SET_GLOBAL_ERROR_DIALOG]:({commit, state}, status)=>{
    commit('SET_GLOBAL_ERROR_DIALOG', status);
  }
  */
};
