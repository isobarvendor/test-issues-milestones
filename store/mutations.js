

//shared store are concatenated to the central store
//TO-DO: manage the shared store seperately by module
export default {

  SET_LOGIN_ACCOUNT:(state, data)=>{
        state.login = data;
    },

  SET_TOKEN:(state, data)=>{
      state.token = data;
  },
  SET_ERROR:(state, data)=>{
    state.errorLogin = data;
},
  SET_LIST_PRIZE:(state, data)=>{
    state.listPrize = data;
  },
  SET_LIST_WALLET:(state, data)=>{
    state.listWallet = data;
  },
  SET_FILE_AMAZON:(state, data)=>{
    state.fileUploaded = data;
  },
  SET_CONFIG:(state,data)=>{
    state.config = data
  },
  SET_CMS_CONTENT:(state,data)=>{
    state.CMSContent = data
  }



};
