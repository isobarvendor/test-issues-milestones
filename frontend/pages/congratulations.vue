<template>
  <div v-if="dataStatus.status == 200" id="main" class="congratulations">
    <div class="wrapper">
      <div class="container">
        <Congrats />
      </div>

    </div>
  </div>
  <div v-else-if="dataStatus.status >= 500">
    Status: {{dataStatus.status}}
    <br />
    {{dataStatus.message}}
  </div>
</template>

<script>
import Congrats from '@/components/Congrats'
export default {
  data(){
    return{
      dataStatus:{},
      data:null,
    }
  },
  head() {
    return {
      title: "Coke Campaign Title",
      meta: [
        {
          hid: "description",
          name: "description",
          content: "Home page description"
        }
      ],
      noscript: [],
      script: [
        {
          hid: "anime",
          src: "/js/anime.min.js"
        }
      ],
      css: []
    };
  },
  created() {
    this.fetchData();
  },
  methods: {
    async fetchData() {
      let result = await this.$axios.get(
        "https://ayo.aircovery.com/cms-api/campaigns"
      );
      this.dataStatus = { status: result.status, message: result.statusText };
      this.data = result.data;
    },
  },
};
</script>

<style lang="scss">
@import '~assets/scss/variables.scss';
  html, body{
    background: $body-bg-color-wallet;
    .congratulations{
        .container{
            position: absolute;
            top: 30%;
        }
    }
    @media only screen and (max-width: 767px) {
      
    }
  }
  
</style>