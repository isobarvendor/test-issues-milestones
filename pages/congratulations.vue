<template>
  <div  class="congratulations">
    <div class="wrapper">
      <div class="container">
        <Congrats />
      </div>

    </div>
  </div>
</template>

<script>
import Congrats from '@/components/Congrats'
import { GET_LIST_WALLET} from '@/store/action_types';
export default {
  data(){
    return{
      dataStatus:{},
      data:null,
    }
  },
  head() {
    return {
      title: "Coke Campaign Title",
      meta: [
        {
          hid: "description",
          name: "description",
          content: "Home page description"
        }
      ],
      noscript: [],
      /*script: [
        {
          hid: "anime",
          src: "/js/anime.min.js"
        }
      ],*/
      css: []
    };
  },
  methods: {
     getListWallet(){
          this.$store.dispatch(GET_LIST_WALLET)
                .then((response)=>{
                })
                .catch((error) =>{
                  if(error.response && error.response.data.status=='401'){
                    localStorage.clear();
                  }
                })


     }
  },
  mounted(){
     this.getListWallet();
  }
};
</script>

<style lang="scss">
@import '~assets/scss/variables.scss';
  html, body{
    background: $body-bg-color-wallet;
    .congratulations{
        .container{
            position: absolute;
            top: 30%;
        }
    }
    @media only screen and (max-width: 767px) {

    }
  }

</style>
