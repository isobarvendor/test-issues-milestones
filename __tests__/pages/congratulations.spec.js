import { mount,  createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router'

import congratulations from '@/pages/congratulations';

import Congrats from '@/components/Congrats'


import {mockCMSConfig,mockCMSContent} from "@/__mock__/mock";


import actions from '@/store/actions';
import mutations from '@/store/mutations';


let localVue, store,router;


  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(VueRouter);
    router = new VueRouter();
    store = new Vuex.Store({
      getters: {
        getCMSContent() {
          return mockCMSContent;
        },
        getLoginAccount(){
          return {
            name:"ron",
            email:"ron@dsw.com",
            imageUrl:"tes"
          }
        },
        getCMSConfig(){
          return mockCMSConfig[0];
        },

      },
      state:{
        redeemPrize:{
          redeemPrize:{
            imgUrl:'image'
          }
        }
      },
      actions,
      mutations
    });
  });

  const factory = () => {
    return mount(congratulations, {

      localVue, store,router
    });
  };



describe('congratulations', () => {


  test('renders a Congrats', () => {
    const wrapper = factory();
    const component= wrapper.findComponent(Congrats);
    expect(component.exists()).toBeTruthy();
  });


  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

});
