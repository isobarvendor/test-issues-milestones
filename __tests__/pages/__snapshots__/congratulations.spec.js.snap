// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`congratulations renders properly 1`] = `
<div class="congratulations">
  <div class="wrapper">
    <div class="container">
      <div></div>
    </div>
  </div>
</div>
`;
